core/src/components/radio-group/test/supporting-text/radio-group.e2e.ts

```diff
@@ -0,0 +1,247 @@
+        <ion-radio-group value="1" helper-text="Helper text" error-text="Error text">
+          <ion-radio value="1">Label</ion-radio>
+          <ion-radio value="2">Label</ion-radio>
+          <ion-radio value="3">Label</ion-radio>
+        </ion-radio-group>
+      `,
+        config
+      );
+
+      const radioGroup = page.locator('ion-radio-group');
+
+      await expect(radioGroup).not.toHaveAttribute('aria-invalid');
+    });
+    test('radio group should not have aria-describedby attribute when no hint or error text is present', async ({
+      page,
+    }) => {
+      await page.setContent(
+        `
+        <ion-radio-group value="1">
+          <ion-radio value="1">Label</ion-radio>
+          <ion-radio value="2">Label</ion-radio>
+          <ion-radio value="3">Label</ion-radio>
+        </ion-radio-group>
+      `,
+        config
+      );
+
+      const radioGroup = page.locator('ion-radio-group');
+
+      await expect(radioGroup).not.toHaveAttribute('aria-describedby');
+    });
+  });
+});
+
+/**
+ * Rendering is different across modes
+ */
+configs({ modes: ['ios', 'md'], directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('radio-group: supporting text rendering'), () => {
+    test('should not have visual regressions when rendering helper text', async ({ page }) => {
+      await page.setContent(
+        `
+        <ion-radio-group value="1" helper-text="Helper text">
+          <ion-radio value="1">Label</ion-radio>
+          <ion-radio value="2">Label</ion-radio>
+          <ion-radio value="3">Label</ion-radio>
+        </ion-radio-group>
+      `,
+        config
+      );
+
+      const radioGroup = page.locator('ion-radio-group');
+      await expect(radioGroup).toHaveScreenshot(screenshot(`radio-group-helper-text`));
+    });
+
+    test('should not have visual regressions when rendering error text', async ({ page }) => {
+      await page.setContent(
+        `
+        <ion-radio-group value="1" class="ion-invalid ion-touched" error-text="Error text">
+          <ion-radio value="1">Label</ion-radio>
+          <ion-radio value="2">Label</ion-radio>
+          <ion-radio value="3">Label</ion-radio>
+        </ion-radio-group>
+      `,
+        config
+      );
+
+      const radioGroup = page.locator('ion-radio-group');
+      await expect(radioGroup).toHaveScreenshot(screenshot(`radio-group-error-text`));
+    });
+  });
+});
+
+/**
+ * Customizing supporting text is the same across modes and directions
+ */
+configs({ modes: ['md'], directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('radio group: supporting text customization'), () => {
+    test('should not have visual regressions when rendering helper text with custom css', async ({ page }) => {
+      await page.setContent(
+        `
+        <style>
+          .radio-group-top {
+            font-size: 20px;
+          }
+
+          .radio-group-top .helper-text {
```

Comment:
Overriding helper and error text will require more specificity since this component is not a shadow component. This is something to keep in mind when documenting it.